packages/grafana-data/src/transformations/transformers/groupBy.ts

```diff
@@ -75,64 +79,19 @@ export const groupByTransformer: DataTransformerInfo<GroupByTransformerOptions>
         const processed: DataFrame[] = [];
 
         for (const frame of data) {
-          const groupByFields: Field[] = [];
-
-          for (const field of frame.fields) {
-            if (shouldGroupOnField(field, options)) {
-              groupByFields.push(field);
-            }
-          }
```

Comment:
It's a tricky situation. On one hand, if we don't refactor we can't reuse the code. But if we do refactor we have the possibility of introducing bugs. I'll give it a thorough run through (and poke at @leeoniya to do the same since he requested the refactor 😆 )